judge/forms.py

```diff
@@ -64,10 +64,16 @@ def clean(self):
     def __init__(self, *args, **kwargs):
         user = kwargs.pop('user', None)
         super(ProfileForm, self).__init__(*args, **kwargs)
-        if not user.has_perm('judge.edit_all_organization'):
-            self.fields['organizations'].queryset = Organization.objects.filter(
+        if not Organization.objects.exists():
+            del self.fields['organizations']
+        elif not user.has_perm('judge.edit_all_organization'):
+            filtered_organizations = Organization.objects.filter(
                 Q(is_open=True) | Q(id__in=user.profile.organizations.all()),
             )
+            if not filtered_organizations.exists():
```

Comment:
This hits the database a lot of unnecessary times. Perhaps we can `del self.fields['organizations']` by directly checking `self.fields['organizations'].queryset` (Assuming we _should_ be `del` in the first place...)

Reply:
Revised: changed `exists()` check to be directly on `self.fields['organizations'].queryset` and replaced `del self.fields['organizations']` with `self.fields['organizations'].pop('organizations')`